cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -216,11 +216,22 @@ private boolean matches(String[] actual, String... expected) {
     return true;
   }
 
+  /**
+   * Determines if the actual ends with expected. If expected contains '?' at the end the the last element of actual
+   * is ignored as ? is considered to match one element regardless of what it is
+   */
```

Comment:
this method can be static and it should have its own unit test

Reply:
Done.